pkgs/development/libraries/librsvg/default.nix

```diff
@@ -1,32 +1,58 @@
-{ lib, stdenv, fetchurl, pkg-config, glib, gdk-pixbuf, pango, cairo, libxml2
-, bzip2, libintl, darwin, rustc, cargo, gnome3
-, vala, gobject-introspection }:
+{ lib
+, stdenv
+, fetchurl
+, pkg-config
+, glib
+, gdk-pixbuf
+, pango
+, cairo
+, libxml2
+, bzip2
+, libintl
+, darwin
+, rustc
+, cargo
+, gnome3
+, vala
+, gobject-introspection
+, nixosTests
+}:
 
-let
-  pname = "librsvg";
-  version = "2.50.1";
-in
 stdenv.mkDerivation rec {
-  name = "${pname}-${version}";
-
-  src = fetchurl {
-    url = "mirror://gnome/sources/${pname}/${lib.versions.majorMinor version}/${name}.tar.xz";
-    sha256 = "02csvx2nzygh8kyal2qiy3y6xb7d52vszxxr37dzav704a9pkncv";
-  };
+  pname = "librsvg";
+  version = "2.50.4";
 
   outputs = [ "out" "dev" "installedTests" ];
 
-  buildInputs = [ libxml2 bzip2 pango libintl ]
-    ++ lib.optionals stdenv.isDarwin [ darwin.libobjc ];
-
-  NIX_LDFLAGS = if stdenv.isDarwin then "-lobjc" else null;
+  src = fetchurl {
+    url = "mirror://gnome/sources/${pname}/${lib.versions.majorMinor version}/${pname}-${version}.tar.xz";
+    sha256 = "18s0c5bqj88gj5lf62mfprvhwfibnkrxi6h2r2si0b4p9dxi1dis";
+  };
 
-  propagatedBuildInputs = [ glib gdk-pixbuf cairo ];
+  nativeBuildInputs = [
+    pkg-config
+    rustc
+    cargo
+    vala
+    gobject-introspection
+  ] ++ lib.optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [
```

Comment:
darwin shouldn't be used as an input.

Reply:
Can you explain what harm it does?

Reply:
No harm per se, but the same reason you don't use `pkgs` as an input. Instead use `ApplicationServices` as an input directly.

Reply:
> Can you explain what harm it does?

It is harder to override it.

Reply:
Done. Could you please check if adding the Foundation was sufficient?